react-app/src/components/SearchBar.js

```diff
@@ -42,25 +42,24 @@ class SearchBar extends Component {
     }
   }
 
+  search = value => {
+    this.previous_input = value;
```

Comment:
What is the purpose of the `previous_input` property?
Why not store the input's value in state?

```js
this.setState({ value });

// then in request():

.get(baseurl + this.tokenize_query(this.state.value))